util/db/db.go

```diff
@@ -153,8 +155,16 @@ func (db *db) GetApplicationControllerReplicas() int {
 	// get the replicas from application controller deployment, if the application controller deployment does not exist, check for environment variable
 	applicationControllerName := env.StringFromEnv(common.EnvAppControllerName, common.DefaultApplicationControllerName)
 	appControllerDeployment, err := db.kubeclientset.AppsV1().Deployments(db.settingsMgr.GetNamespace()).Get(context.Background(), applicationControllerName, metav1.GetOptions{})
-
-	if err == nil && appControllerDeployment.Spec.Replicas != nil {
+	if err != nil {
+		if kubeerrors.IsNotFound(err) {
+			appControllerDeployment = nil
```

Comment:
We should set `appControllerDeployment = nil` irrespective of whether we find the `appControllerDeployment` or not. If we get an error, we basically did not find the deployment and `appControllerDeployment` will be set as an empty object.